client/insert.go

```diff
@@ -372,7 +377,7 @@ func (c *GrpcClient) Upsert(ctx context.Context, collName string, partitionName
 		CollectionName: collName,
 		PartitionName:  partitionName,
 	}
-	if req.PartitionName == "" {
+	if req.PartitionName == "" && !hasPartitionKey {
 		req.PartitionName = "_default" // use default partition
 	}
```

Comment:
As mentioned here
https://github.com/milvus-io/milvus-sdk-go/issues/571#issuecomment-1704938617
this default partition name fulfilling behavior could be removed even partition key is not enabled